collector/indices.go

```diff
@@ -0,0 +1,182 @@
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	defaultIndexLabels      = []string{"cluster", "index"}
+	defaultIndexLabelValues = func(clusterName string, indexName string) []string {
+		return []string{clusterName, indexName}
+	}
+)
+
+type indexMetric struct {
+	Type   prometheus.ValueType
+	Desc   *prometheus.Desc
+	Value  func(indexStats IndexStatsIndexResponse) float64
+	Labels func(clusterName string, indexName string) []string
+}
+
+type Indices struct {
+	logger        log.Logger
+	client        *http.Client
+	url           *url.URL
+	all           bool
+	exportIndices bool
```

Comment:
please move the bool controlling if indices metrics should be scraped out of the collector. There is no reason registering the metrics via `Describe()` and then don't collect them. See my other comment in main.go